crates/router/src/core/unified_connector_service.rs

```diff
@@ -55,6 +55,16 @@ pub mod transformers;
 // Re-export webhook transformer types for easier access
 pub use transformers::WebhookTransformData;
 
+/// Type alias for the complex return type used by unified connector service response handlers
+type UnifiedConnectorServiceResult = CustomResult<
+    (
+        Option<AttemptStatus>,
+        Result<PaymentsResponseData, ErrorResponse>,
+        u16,
+    ),
+    UnifiedConnectorServiceError,
+>;
```

Comment:
Since we know that Attempt status will only be returned during Ok response put it as a part of inner result. So In case of Err, it will not be retuned. And there is no need to convert from grpc::PaymentStatus to Option<AttemptStatus>.


This needs to be handles appropriately in the response handler function.
```suggestion
type UnifiedConnectorServiceResult = CustomResult<
    (
        Result<(PaymentsResponseData, AttemptStatus), ErrorResponse>,
        u16,
    ),
    UnifiedConnectorServiceError,
>;
```

Reply:
Attempt status returned in both cases

Reply:
Attempt status is there as an optional field in `struct ErrorResponse`.